$NetBSD: patch-meson.build,v 1.2 2024/12/18 13:28:05 hauke Exp $

While NetBSD has libwrap, the meson code doesn't find it. The test
cond still succeeds, so add a branch that does not require the
library.

Install at.h on Linux, we do it on NetBSD, too
 
--- meson.build.orig	2024-11-15 06:52:12.000000000 +0000
+++ meson.build
@@ -2010,7 +2010,11 @@ int main(void) {
 }
 '''
 
-    have_tcpwrap = wrap.found() and cc.links(tcpwrap_code, args: '-lwrap')
+    # *BSD have tcpwrappers support in libc
+    have_tcpwrap = wrap.found() and cc.links(tcpwrap_code)
+    if not have_tcpwrap
+	have_tcpwrap = wrap.found() and cc.links(tcpwrap_code, args: '-lwrap')
+    endif
     if not have_tcpwrap
         have_tcpwrap = cc.links(tcpwrap_code, args: ['-lwrap', '-lnsl'])
     endif
@@ -2077,7 +2081,6 @@ if host_os.contains('freebsd')
     cdata.set('FREEBSD', 1)
     cdata.set('OPEN_NOFOLLOW_ERRNO', 'EMLINK')
 elif host_os.contains('linux')
-    use_glibc_at_header = cc.has_header('netatalk/at.h')
     cdata.set('_GNU_SOURCE', 1)
     if cc.compiles(
         '''
